tracer/test/benchmarks/Benchmarks.Trace/TraceAnnotationsBenchmark.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class TraceAnnotationsBenchmark
+    {
+        private static readonly RuntimeMethodHandle MethodHandle;
+        private static readonly RuntimeTypeHandle TypeHandle;
+
+        static TraceAnnotationsBenchmark()
+        {
+            var settings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false
+            };
+
+            Tracer.UnsafeSetTracerInstance(new Tracer(settings, new DummyAgentWriter(), null, null, null));
+
+            var targetMethod = typeof(TraceAnnotationsBenchmark).GetMethod("InstrumentedMethod");
+            MethodHandle = targetMethod.MethodHandle;
+            TypeHandle = targetMethod.DeclaringType.TypeHandle;
+
+            var bench = new TraceAnnotationsBenchmark();
+            bench.RunOnMethodBegin();
+        }
+
+        [Benchmark]
+        public CallTargetState RunOnMethodBegin()
+        {
+            CallTargetState returnValue = default;
+
+            for (int i = 0; i < 25; i++)
+            {
+                returnValue = TraceAnnotationsIntegration.OnMethodBegin<object>(null, MethodHandle, TypeHandle);
+            }
+
+            return returnValue;
+        }
```

Comment:
```suggestion
        public CallTargetReturn RunOnMethodBegin()
        {
            var state = TraceAnnotationsIntegration.OnMethodBegin<object>(null, MethodHandle, TypeHandle);
            return TraceAnnotationsIntegration.OnMethodEnd<object>(null, null, state);
        }
```

Reply:
Thanks for the suggestion, implemented in b0acf279f8ca63ccaf48f528964ab92366ed94a6

I wanted to avoid running the disposal code in the hot path, but the way I wrote the benchmark and (no) clean-up code led to issues. It should be fine to run both the OnMethodBegin and OnMethodEnd code in the benchmark as that entire code will be new overhead for otherwise uninstrumented application code